src/player.js

```diff
@@ -1006,6 +1035,8 @@ export default class Player extends FakeEventTarget {
         if (track.language === OFF) {
           this.hideTextTrack();
           this._playbackAttributesState.textLanguage = OFF;
+        } else if (track.external && !this._config.playback.useNativeTextTrack) {
+          this._externalCaptionsHandler.selectTextTrack(track);
```

Comment:
need to call `hideTextTrack` on engine when selecting external, and vide versa when calling the native select, hide the external